### ISSUE TITLE
Add keyboard shortcuts for desktop apps

### ISSUE THREAD

**Issue by devstra** [streetwriters/notesnook#1014]
### What problem are you facing?

There are no/barely any keyboard shortcuts in the desktop apps and in the note editor. This makes writting notes slow & unproductive especially when coming from another note-taking app who has shortcuts.

### What's the solution you'd like to see in Notesnook?

It would be great to have keyboard shortcuts like many of the other note-taking apps available since it greatly increases productivity and can make it easier to switch from another app.

Some notable shortcuts I'd like to see:
- search (shortcut displays search bar & focuses input to search my notes at any point)
- editor shortcuts (adding links, highlighting, header, lists etc) not necessarily for all styling options but at least the main ones would be nice.

### Additional context

Anyone, feel free to add your shortcut ideas here.

Could be a dup of #105 ?

**Comment by devstra** [comment]
I'd be willing to work on this issue btw, if it's inline with your roadmap and if I had a bit of help since I don't know the codebase yet.

**Comment by short-y** [comment]
There are quite a few shortcuts in the code but they're not documented very well, if at all. For instance ALT-D inserts the date, ALT-T inserts the time, MOD-ALT-D inserts time and date, MOD-ALT-C toggles code block. But I agree therre could be many more shortcuts including ALT-/ or ALT-? to show the current shortcuts.\\

Supposedly the package used for the editor has shortcuts, too although I'm not sure what they are. Certainly, by experimenting, I fidc that CTL-B toggles BOLD, CTL-I toggles italics

**Comment by devstra** [comment]
I was also able to find the same shortcuts but trying out random combinations on my keyboard (yeah lol) and you make a good point about keyboard shortcuts, that should be explained somewhere like a help menu for example.

**Comment by devstra** [comment]
https://tiptap.dev/api/keyboard-shortcuts#predefined-keyboard-shortcuts

Not all of the shortcuts explained above work in Notesnook but there are a few more shortcuts than I thought.

**Comment by McBaumwolle** [comment]
I'd love to see something like this in Notesnook! Especially like that command search in Notion or VS Code where you can search anything. 
My suggestion would be a simple command editor or presets for like "selecting the current color" where I select text with the keyboard for example and then "Alt+P" and it gets set to the current color.

**Comment by ahusby** [comment]
Yeah, I can't use this kind of app with mouse. This is a deal-breaker for me. Writing notes is a keyboard based activity, my hands should always be on the keyboard. Reaching for the mouse disturbs the flow of thought. The more I work with computers the less I use the mouse. While I am doing my thing I should be able to jot down a note quickly almost without thinking about it, without switching context and making the note taking my primary task.

**Comment by Gahwa-Gahwa** [comment]
+1 I second this. We need keyboard shortcuts, using mouse for everything is annoying and breaks my workflow. Even simple things like creating a new note, entering focus mode or aligning text can only be done with mouse, ..

**Comment by Gahwa-Gahwa** [comment]
+1 I second this. We need keyboard shortcuts, using mouse for everything is annoying and breaks my workflow. Even simple things like creating a new note, entering focus mode or aligning text can only be done with mouse.

**Comment by dep** [comment]
Some essentials for me:

- `cmd-n`, new note (maybe with a `cmd-t` alias)
- `cmd-w`, close tab
- `cmd-del`, remove active note

Down the road would love to see `cmd-k` trigger a command palette which would allow me to typeahead common actions (eg. a box pops up where I can type things like "publish", "lock", etc)

**Comment by Havock94** [comment]
I'd also propose a shortcut to lock the app manually

**Comment by bravesfan21** [comment]
I would love to have a highlight shortcut (ctrl + shift + h) added.

**Comment by distefam** [comment]
I just signed up for Notesnook and was trying to figure out how to do things with my keyboard, which sent me here. This is a bit of a dealbreaker for me too and I think I'll hold off on purchasing a premium subscription until this is ready.

**Comment by KitN** [comment]
This might be a dealbreaker for me too.

**Comment by uberjay** [comment]
One thing that's driving me crazy: in macOS, `control-n` creates a new note instead of moving to the next line.

In the process of editing a single note I accidentally create a new blank note approximately 50 times. 😩 

Relatedly: `control-p` opens a search/command palette instead of moving to the previous line. It seems like the only shortcut (that I use frequently 😁) that works is `control-b`, for moving backwards by one character.

Reference: [macOS text-editing shortcuts](https://support.apple.com/en-us/102650#text).

**Comment by Polygon1556** [comment]
Is no one talking about implementing VIM like text editing on desktop? This would make using this app so much more efficient for me.

**Comment by Froth8000** [comment]
Please, this is hugely important to me. I'm moving away from UpNote and the two things i LOVE that they do really well are the collapsible sections (NN does these but the formatting isn't as clean or intuitive) and the keyboard-shortcut-on-hover. I just want to be able to flow without having to touch the mouse. If you can implement this, i'm all in.

**Comment by 01zulfi** [comment]
many keyboard shortcuts were added over the course of last few months, full list here https://help.notesnook.com/keyboard-shortcuts

please open a new issue for a specific one
